distributedlog-core/src/main/java/org/apache/distributedlog/EnvelopedEntryWriter.java

```diff
@@ -56,35 +60,33 @@
     }
 
     private final String logName;
-    private final Buffer buffer;
+    private final ByteBuf buffer;
+    private ByteBuf finalizedBuffer = null;
     private final LogRecord.Writer writer;
     private final List<WriteRequest> writeRequests;
     private final boolean envelopeBeforeTransmit;
     private final CompressionCodec.Type codec;
-    private final StatsLogger statsLogger;
+    private final int flags;
     private int count = 0;
     private boolean hasUserData = false;
     private long maxTxId = Long.MIN_VALUE;
 
     EnvelopedEntryWriter(String logName,
                          int initialBufferSize,
                          boolean envelopeBeforeTransmit,
-                         CompressionCodec.Type codec,
-                         StatsLogger statsLogger) {
+                         CompressionCodec.Type codec) {
         this.logName = logName;
-        this.buffer = new Buffer(initialBufferSize * 6 / 5);
-        this.writer = new LogRecord.Writer(new DataOutputStream(buffer));
+        this.buffer = PooledByteBufAllocator.DEFAULT.buffer(
+                Math.min(Math.max(initialBufferSize * 6 / 5, HEADER_LENGTH), MAX_LOGRECORDSET_SIZE),
```

Comment:
why this *6/5 ?

Reply:
I don't remember what was exactly the reason that we used *6/5. We can probably remove it.